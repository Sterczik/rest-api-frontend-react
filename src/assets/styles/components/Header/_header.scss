.header {
  flex-grow: 1;
  &__typo {
    flex-grow: 1;
  }
  &__link {
    color: white;
    text-decoration: none;
  }
}
