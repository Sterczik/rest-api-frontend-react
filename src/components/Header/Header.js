import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import AccountCircle from '@material-ui/icons/AccountCircle';

import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { authActions } from '../../containers/App/auth/actions';

export class Header extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      anchorEl: null
    };

    this.handleClick = this.handleClick.bind(this);
    this.handleClose = this.handleClose.bind(this);
    this.logout = this.logout.bind(this);
  }

  handleClick(event) {
    this.setState({ anchorEl: event.currentTarget });
  }

  handleClose() {
    this.setState({ anchorEl: null });
  }

  logout() {
    this.handleClose();
    this.props.logout();
  }

  render() {
    return (
      <div className="header">
        <AppBar position="static">
          <Toolbar>
            <Typography className="header__typo" variant="headline" color="inherit">
              <Link to="/todos" className="header__link">
                Starter
              </Link>
            </Typography>
            <IconButton
              aria-owns={this.state.anchorEl ? 'menu-appbar' : undefined}
              aria-haspopup="true"
              onClick={this.handleClick}
              color="inherit"
            >
              <AccountCircle />
            </IconButton>
            <Menu
              id="simple-menu"
              anchorEl={this.state.anchorEl}
              open={Boolean(this.state.anchorEl)}
              onClose={this.handleClose}
            >
              <MenuItem className="header__anchor" component={Link} to="/todos">Todos</MenuItem>
              <MenuItem className="header__anchor" component={Link} to="/my-account">My account</MenuItem>
              <MenuItem className="header__anchor" onClick={this.logout}>Logout</MenuItem>
            </Menu>
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  logout: () => {
    dispatch(authActions.logout());
  }
});

export default connect(undefined, mapDispatchToProps)(Header);
