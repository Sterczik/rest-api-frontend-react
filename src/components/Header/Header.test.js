import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

let wrapper;
let logout;

beforeEach(() => {
  logout = jest.fn();
  wrapper = shallow(<Header
    logout={logout}
  />);
});

test('should render Header correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle logout', () => {
  wrapper.find('.header__anchor').at(2).simulate('click');
  expect(logout).toHaveBeenCalled();
});
