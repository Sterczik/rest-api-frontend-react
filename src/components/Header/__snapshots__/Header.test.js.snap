// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    logout={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(AppBar)
        position="static"
      >
        <WithStyles(Toolbar)>
          <WithStyles(Typography)
            className="header__typo"
            color="inherit"
            variant="headline"
          >
            <Link
              className="header__link"
              replace={false}
              to="/todos"
            >
              Starter
            </Link>
          </WithStyles(Typography)>
          <WithStyles(IconButton)
            aria-haspopup="true"
            color="inherit"
            onClick={[Function]}
          >
            <pure(AccountCircleIcon) />
          </WithStyles(IconButton)>
          <WithStyles(Menu)
            anchorEl={null}
            id="simple-menu"
            onClose={[Function]}
            open={false}
          >
            <WithStyles(MenuItem)
              className="header__anchor"
              component={[Function]}
              to="/todos"
            >
              Todos
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              className="header__anchor"
              component={[Function]}
              to="/my-account"
            >
              My account
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              className="header__anchor"
              onClick={[Function]}
            >
              Logout
            </WithStyles(MenuItem)>
          </WithStyles(Menu)>
        </WithStyles(Toolbar)>
      </WithStyles(AppBar)>,
      "className": "header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Toolbar)>
          <WithStyles(Typography)
            className="header__typo"
            color="inherit"
            variant="headline"
          >
            <Link
              className="header__link"
              replace={false}
              to="/todos"
            >
              Starter
            </Link>
          </WithStyles(Typography)>
          <WithStyles(IconButton)
            aria-haspopup="true"
            color="inherit"
            onClick={[Function]}
          >
            <pure(AccountCircleIcon) />
          </WithStyles(IconButton)>
          <WithStyles(Menu)
            anchorEl={null}
            id="simple-menu"
            onClose={[Function]}
            open={false}
          >
            <WithStyles(MenuItem)
              className="header__anchor"
              component={[Function]}
              to="/todos"
            >
              Todos
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              className="header__anchor"
              component={[Function]}
              to="/my-account"
            >
              My account
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              className="header__anchor"
              onClick={[Function]}
            >
              Logout
            </WithStyles(MenuItem)>
          </WithStyles(Menu)>
        </WithStyles(Toolbar)>,
        "position": "static",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              className="header__typo"
              color="inherit"
              variant="headline"
            >
              <Link
                className="header__link"
                replace={false}
                to="/todos"
              >
                Starter
              </Link>
            </WithStyles(Typography)>,
            <WithStyles(IconButton)
              aria-haspopup="true"
              color="inherit"
              onClick={[Function]}
            >
              <pure(AccountCircleIcon) />
            </WithStyles(IconButton)>,
            <WithStyles(Menu)
              anchorEl={null}
              id="simple-menu"
              onClose={[Function]}
              open={false}
            >
              <WithStyles(MenuItem)
                className="header__anchor"
                component={[Function]}
                to="/todos"
              >
                Todos
              </WithStyles(MenuItem)>
              <WithStyles(MenuItem)
                className="header__anchor"
                component={[Function]}
                to="/my-account"
              >
                My account
              </WithStyles(MenuItem)>
              <WithStyles(MenuItem)
                className="header__anchor"
                onClick={[Function]}
              >
                Logout
              </WithStyles(MenuItem)>
            </WithStyles(Menu)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Link
                className="header__link"
                replace={false}
                to="/todos"
              >
                Starter
              </Link>,
              "className": "header__typo",
              "color": "inherit",
              "variant": "headline",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Starter",
                "className": "header__link",
                "replace": false,
                "to": "/todos",
              },
              "ref": null,
              "rendered": "Starter",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-haspopup": "true",
              "aria-owns": undefined,
              "children": <pure(AccountCircleIcon) />,
              "color": "inherit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "anchorEl": null,
              "children": Array [
                <WithStyles(MenuItem)
                  className="header__anchor"
                  component={[Function]}
                  to="/todos"
                >
                  Todos
                </WithStyles(MenuItem)>,
                <WithStyles(MenuItem)
                  className="header__anchor"
                  component={[Function]}
                  to="/my-account"
                >
                  My account
                </WithStyles(MenuItem)>,
                <WithStyles(MenuItem)
                  className="header__anchor"
                  onClick={[Function]}
                >
                  Logout
                </WithStyles(MenuItem)>,
              ],
              "id": "simple-menu",
              "onClose": [Function],
              "open": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Todos",
                  "className": "header__anchor",
                  "component": [Function],
                  "to": "/todos",
                },
                "ref": null,
                "rendered": "Todos",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "My account",
                  "className": "header__anchor",
                  "component": [Function],
                  "to": "/my-account",
                },
                "ref": null,
                "rendered": "My account",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Logout",
                  "className": "header__anchor",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Logout",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(AppBar)
          position="static"
        >
          <WithStyles(Toolbar)>
            <WithStyles(Typography)
              className="header__typo"
              color="inherit"
              variant="headline"
            >
              <Link
                className="header__link"
                replace={false}
                to="/todos"
              >
                Starter
              </Link>
            </WithStyles(Typography)>
            <WithStyles(IconButton)
              aria-haspopup="true"
              color="inherit"
              onClick={[Function]}
            >
              <pure(AccountCircleIcon) />
            </WithStyles(IconButton)>
            <WithStyles(Menu)
              anchorEl={null}
              id="simple-menu"
              onClose={[Function]}
              open={false}
            >
              <WithStyles(MenuItem)
                className="header__anchor"
                component={[Function]}
                to="/todos"
              >
                Todos
              </WithStyles(MenuItem)>
              <WithStyles(MenuItem)
                className="header__anchor"
                component={[Function]}
                to="/my-account"
              >
                My account
              </WithStyles(MenuItem)>
              <WithStyles(MenuItem)
                className="header__anchor"
                onClick={[Function]}
              >
                Logout
              </WithStyles(MenuItem)>
            </WithStyles(Menu)>
          </WithStyles(Toolbar)>
        </WithStyles(AppBar)>,
        "className": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Toolbar)>
            <WithStyles(Typography)
              className="header__typo"
              color="inherit"
              variant="headline"
            >
              <Link
                className="header__link"
                replace={false}
                to="/todos"
              >
                Starter
              </Link>
            </WithStyles(Typography)>
            <WithStyles(IconButton)
              aria-haspopup="true"
              color="inherit"
              onClick={[Function]}
            >
              <pure(AccountCircleIcon) />
            </WithStyles(IconButton)>
            <WithStyles(Menu)
              anchorEl={null}
              id="simple-menu"
              onClose={[Function]}
              open={false}
            >
              <WithStyles(MenuItem)
                className="header__anchor"
                component={[Function]}
                to="/todos"
              >
                Todos
              </WithStyles(MenuItem)>
              <WithStyles(MenuItem)
                className="header__anchor"
                component={[Function]}
                to="/my-account"
              >
                My account
              </WithStyles(MenuItem)>
              <WithStyles(MenuItem)
                className="header__anchor"
                onClick={[Function]}
              >
                Logout
              </WithStyles(MenuItem)>
            </WithStyles(Menu)>
          </WithStyles(Toolbar)>,
          "position": "static",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                className="header__typo"
                color="inherit"
                variant="headline"
              >
                <Link
                  className="header__link"
                  replace={false}
                  to="/todos"
                >
                  Starter
                </Link>
              </WithStyles(Typography)>,
              <WithStyles(IconButton)
                aria-haspopup="true"
                color="inherit"
                onClick={[Function]}
              >
                <pure(AccountCircleIcon) />
              </WithStyles(IconButton)>,
              <WithStyles(Menu)
                anchorEl={null}
                id="simple-menu"
                onClose={[Function]}
                open={false}
              >
                <WithStyles(MenuItem)
                  className="header__anchor"
                  component={[Function]}
                  to="/todos"
                >
                  Todos
                </WithStyles(MenuItem)>
                <WithStyles(MenuItem)
                  className="header__anchor"
                  component={[Function]}
                  to="/my-account"
                >
                  My account
                </WithStyles(MenuItem)>
                <WithStyles(MenuItem)
                  className="header__anchor"
                  onClick={[Function]}
                >
                  Logout
                </WithStyles(MenuItem)>
              </WithStyles(Menu)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Link
                  className="header__link"
                  replace={false}
                  to="/todos"
                >
                  Starter
                </Link>,
                "className": "header__typo",
                "color": "inherit",
                "variant": "headline",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Starter",
                  "className": "header__link",
                  "replace": false,
                  "to": "/todos",
                },
                "ref": null,
                "rendered": "Starter",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-haspopup": "true",
                "aria-owns": undefined,
                "children": <pure(AccountCircleIcon) />,
                "color": "inherit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "anchorEl": null,
                "children": Array [
                  <WithStyles(MenuItem)
                    className="header__anchor"
                    component={[Function]}
                    to="/todos"
                  >
                    Todos
                  </WithStyles(MenuItem)>,
                  <WithStyles(MenuItem)
                    className="header__anchor"
                    component={[Function]}
                    to="/my-account"
                  >
                    My account
                  </WithStyles(MenuItem)>,
                  <WithStyles(MenuItem)
                    className="header__anchor"
                    onClick={[Function]}
                  >
                    Logout
                  </WithStyles(MenuItem)>,
                ],
                "id": "simple-menu",
                "onClose": [Function],
                "open": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Todos",
                    "className": "header__anchor",
                    "component": [Function],
                    "to": "/todos",
                  },
                  "ref": null,
                  "rendered": "Todos",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "My account",
                    "className": "header__anchor",
                    "component": [Function],
                    "to": "/my-account",
                  },
                  "ref": null,
                  "rendered": "My account",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Logout",
                    "className": "header__anchor",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Logout",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
